packages/wallet/src/redux/reducers/index.ts

```diff
@@ -1,133 +1,56 @@
-import {
-  WalletState,
-  INITIALIZING,
-  OPENING,
-  FUNDING,
-  RUNNING,
-  CHALLENGING,
-  RESPONDING,
-  WITHDRAWING,
-  CLOSING,
-  waitForLogin,
-  approveConclude,
-  ApproveConclude,
-  acknowledgeConclude,
-  AcknowledgeConclude,
-} from '../states';
+import { WalletState, INITIALIZING, waitForLogin, WALLET_INITIALIZED } from '../states';
 
 import { initializingReducer } from './initializing';
-import { openingReducer } from './opening';
-import { fundingReducer } from './funding';
-import { runningReducer } from './running';
-import { challengingReducer } from './challenging';
-import { respondingReducer } from './responding';
-import { withdrawingReducer } from './withdrawing';
-import { closingReducer } from './closing';
 import {
   WalletAction,
-  CONCLUDE_REQUESTED,
   MESSAGE_SENT,
   TRANSACTION_SENT_TO_METAMASK,
   DISPLAY_MESSAGE_SENT,
-  COMMITMENT_RECEIVED,
 } from '../actions';
-import { unreachable, ourTurn, validTransition } from '../../utils/reducer-utils';
-import { validCommitmentSignature } from '../../utils/signing-utils';
-import { showWallet } from 'magmo-wallet-client/lib/wallet-events';
-import { CommitmentType } from 'fmg-core';
+import { unreachable } from '../../utils/reducer-utils';
+import { OutboxState, SharedWalletState } from '../states/shared';
+import { initializedReducer } from './initialized';
 
 const initialState = waitForLogin();
 
 export const walletReducer = (
   state: WalletState = initialState,
   action: WalletAction,
 ): WalletState => {
+  const nextOutbox: OutboxState = {};
   if (action.type === MESSAGE_SENT) {
-    state = { ...state, messageOutbox: undefined };
+    nextOutbox.messageOutbox = undefined;
   }
   if (action.type === DISPLAY_MESSAGE_SENT) {
-    state = { ...state, displayOutbox: undefined };
+    nextOutbox.displayOutbox = undefined;
   }
-
   if (action.type === TRANSACTION_SENT_TO_METAMASK) {
-    state = { ...state, transactionOutbox: undefined };
-  }
-
-  const conclusionStateFromOwnRequest = receivedValidOwnConclusionRequest(state, action);
-  if (conclusionStateFromOwnRequest) {
-    return conclusionStateFromOwnRequest;
-  }
-
-  const conclusionStateFromOpponentRequest = receivedValidOpponentConclusionRequest(state, action);
-  if (conclusionStateFromOpponentRequest) {
-    return conclusionStateFromOpponentRequest;
+    nextOutbox.transactionOutbox = undefined;
   }
+  state = sideEffectsReducer(state, nextOutbox);
```

Comment:
@andrewgordstewart If I am understanding the logic correctly, when the wallet is in the `CHANNEL_INITIALIZED` state, the `sideEffectsReducer` will get called twice (once in the `walletReducer` and once in `initializedReducer`)? Is this intended?

Reply:
Also, I think `nextOutbox` might always be an empty object when on this line.